Source/TrieRouteMatcher.swift

```diff
@@ -35,36 +35,42 @@ public struct TrieRouteMatcher: RouteMatcherType {
         var nextComponentId = 1
 
         for route in routes {
-
-            // turn component (string) into an id (integer) for fast comparisons
-            let componentIds = route.path.split("/").map { component -> Int in
-
-                // if it already has a component with the same name, use that id
-                if let id = componentsTrie.findPayload(component.characters) {
+            
+            for method in route.methods {
+                
+                // add the method to the path so it is checked
+                let path = method.description + route.path
```

Comment:
So I tried it that way first thinking I needed it but route.path started with a "/" so I was getting GET//my/path instead of the intended GET/my/path.